lib/builders/dataTypes/listViewDataTypeBuilder.ts

```diff
@@ -0,0 +1,126 @@
+﻿import {DataTypeBuilder} from "./dataTypeBuilder";
+import {ListViewBulkActionPermissionsBuilder, ListViewLayoutBuilder, ListViewPropertyBuilder} from "./listViewBuilder";
+
+export class ListViewDataTypeBuilder extends DataTypeBuilder {
+  pageSize: number;
+  orderBy: string;
+  orderDirection: string;
+  layouts: ListViewLayoutBuilder[];
+  includeProperties: ListViewPropertyBuilder[];
+  bulkActionPermissions: ListViewBulkActionPermissionsBuilder;
+  icon: string;
+  tabName: string;
+  showContentFirst: boolean;
+
+  constructor() {
+    super();
+    this.editorAlias = "Umbraco.ListView";
+    this.editorUiAlias = "Umb.PropertyEditorUi.Collection";
+    this.layouts = [];
+    this.includeProperties = [];
+    this.bulkActionPermissions = new ListViewBulkActionPermissionsBuilder(this);
+  }
+
+  withPageSize(pageSize: number) {
+    this.pageSize = pageSize;
+    return this;
+  }
+
+  withOrderBy(orderBy: string) {
+    this.orderBy = orderBy;
+    return this;
+  }
+
+  withOrderDirection(orderDirection: string) {
+    this.orderDirection = orderDirection;
+    return this;
+  }
+
+  addLayout() {
+    const builder = new ListViewLayoutBuilder(this);
+    this.layouts.push(builder);
+    return builder;
+  }
+
+  addColumnDisplayedProperty() {
+    const builder = new ListViewPropertyBuilder(this);
+    this.includeProperties.push(builder);
+    return builder;
+  }
+
+  addBulkActionPermissions() {
+    const builder = new ListViewBulkActionPermissionsBuilder(this);
+    this.bulkActionPermissions = builder;
+    return builder;
+  }
+
+  withIcon(icon: string) {
+    this.icon = icon;
+    return this;
+  }
+
+  withTabName(tabName: string) {
```

Comment:
Would it be better to change the name to "withAppName" as in GUI, it is "Content app name"?